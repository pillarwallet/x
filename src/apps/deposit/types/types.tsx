export type BalanceInfo = {
  type?: 'BalanceInfo';
  name: string;
  chain: string;
  address: string;
  decimals: number;
  symbol: string;
  logoURI: string;
  balance: string;
};

export type AddedAssets = {
  type?: 'AddedAsset';
  assetType: 'token' | 'nft';
  tokenId?: string;
  chain: string;
  tokenAddress: string;
  balance: number;
};

export type Network =
  | 'ethereum'
  | 'polygon'
  | 'gnosis'
  | 'base'
  | 'bnb smart chain'
  | 'optimism'
  | 'arbitrum';

export type TokenList = {
  address: string;
  chainId: number;
  decimals: string;
  name: string;
  symbol: string;
  logoURI: string;
};

export type BalancesByChain = {
  [chain: string]: BalanceInfo[];
};
