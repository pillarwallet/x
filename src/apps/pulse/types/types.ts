export type TransactionStatusState =
  | 'Starting Transaction'
  | 'Transaction Pending'
  | 'Transaction Complete'
  | 'Transaction Failed';

export type TransactionStep =
  | 'Submitted'
  | 'Pending'
  | 'ResourceLock'
  | 'Completed';

export type StepStatus = 'completed' | 'pending' | 'failed' | 'inactive';

export interface TokenInfo {
  symbol: string;
  name: string;
  logo: string;
  address?: string;
}

export interface SellOffer {
  tokenAmountToReceive: number;
  minimumReceive: number;
}

export interface PayingToken {
  totalUsd: number;
  name: string;
  symbol: string;
  logo: string;
  actualBal: string;
  totalRaw: string;
  chainId: number;
  address: string;
}

export interface TokenDetails {
  symbol: string;
  name: string;
  address: string;
  chainId: number;
  amount: string;
  logo: string;
  type: 'BUY_TOKEN' | 'SELL_TOKEN';
}

export interface BuyModeDetails {
  usdAmount: string;
  payingTokens: PayingToken[];
  totalPayingUsd: number;
}

export interface TransactionStatusConfig {
  icon: string;
  containerClasses: string;
  iconClasses: string;
  color: string;
}

export interface ButtonConfig {
  bgColor: string;
  textColor: string;
  borderColor: string;
  label: string;
}

export interface ProgressStepConfig {
  label: string;
  order: number;
}

export interface TransactionStatusProps {
  closeTransactionStatus: () => void;
  userOpHash: string;
  chainId: number;
  gasFee?: string;
  // Transaction data from PreviewSell/PreviewBuy
  isBuy?: boolean;
  sellToken?: TokenInfo | null;
  buyToken?: TokenInfo | null;
  tokenAmount?: string;
  sellOffer?: SellOffer | null;
  payingTokens?: PayingToken[];
  usdAmount?: string;
  // Externalized polling state
  currentStatus: TransactionStatusState;
  errorDetails: string;
  submittedAt?: Date;
  pendingCompletedAt?: Date;
  blockchainTxHash?: string;
  resourceLockTxHash?: string;
  completedTxHash?: string;
  completedChainId?: number;
  resourceLockChainId?: number;
  resourceLockCompletedAt?: Date;
  isResourceLockFailed?: boolean;
}

export interface TransactionDetailsProps {
  onDone: () => void;
  userOpHash: string;
  chainId: number;
  status: TransactionStatusState;
  // Transaction data from PreviewSell/PreviewBuy
  isBuy?: boolean;
  sellToken?: TokenInfo | null;
  buyToken?: TokenInfo | null;
  tokenAmount?: string;
  sellOffer?: SellOffer | null;
  payingTokens?: PayingToken[];
  usdAmount?: string;
  submittedAt?: Date;
  pendingCompletedAt?: Date;
  resourceLockCompletedAt?: Date;
  txHash?: string;
  gasFee?: string;
  errorDetails?: string;
  resourceLockTxHash?: string;
  completedTxHash?: string;
  resourceLockChainId?: number;
  completedChainId?: number;
  isResourceLockFailed?: boolean;
}

export interface TransactionInfoProps {
  status: TransactionStatusState;
  userOpHash: string;
  txHash?: string;
  chainId: number;
  gasFee?: string;
  completedAt?: Date;
  // Buy-specific fields
  isBuy?: boolean;
  resourceLockTxHash?: string;
  resourceLockChainId?: number;
  completedTxHash?: string;
  completedChainId?: number;
}

export interface ProgressStepProps {
  step: TransactionStep;
  status: StepStatus;
  label: string;
  isLast?: boolean;
  showLine?: boolean;
  lineStatus?: StepStatus;
  timestamp?: string | React.ReactNode;
}

export interface TransactionErrorBoxProps {
  technicalDetails?: string;
}

export interface UseClickOutsideOptions {
  ref: React.RefObject<HTMLElement>;
  callback: () => void;
  condition: boolean;
}

export interface UseKeyboardNavigationOptions {
  onEscape: () => void;
  onEnter?: () => void;
  enabled?: boolean;
}

export interface TechnicalDetails {
  transactionType: 'BUY' | 'SELL';
  transactionHash: string;
  hashType: 'bidHash' | 'userOpHash';
  chainId: number;
  status: TransactionStatusState;
  timestamp: string;
  accountAddress: string;
  token: TokenDetails | null;
  sellOffer: SellOffer | null;
  buyMode: BuyModeDetails | null;
  transactionHashes: {
    [key: string]: string;
  };
  chains: {
    mainChainId: number;
    resourceLockChainId: string | number;
    completedChainId: string | number;
  };
  timestamps: {
    [key: string]: string;
  };
  error: {
    details: string;
    isResourceLockFailed: boolean;
    failureStep: string;
  };
  gas: {
    fee: string;
  };
  stepStatus: {
    [key: string]: StepStatus;
  };
}
