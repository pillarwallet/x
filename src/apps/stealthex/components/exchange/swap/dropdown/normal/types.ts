
import type React from 'react';
import { StructuredCurrency } from '../../../../../type';
import { CurrencyData } from '../../../../../lib/backend/api';

export type ClickCallback = (currency: CurrencyData) => void;

export type AvailableCurrency =
  | {
    availableCurrency?: Omit<StructuredCurrency, 'fiat'> & {
      all?: CurrencyData[];
    };
    type: 'get';
  }
  | {
    availableCurrency?: StructuredCurrency & {
      all?: CurrencyData[];
    };
    type: 'send';
  };

export type NormalDropdownProps = AvailableCurrency & {
  loading?: boolean;
  selectedCurrency: CurrencyData | null;
  searchContext: string;
  fiat?: boolean;
  onClick?: ClickCallback;
  components?: {
    NotFound?: React.FC<React.PropsWithChildren>;
    Group?: React.FC<React.PropsWithChildren>;
    Item?: React.FC<
      React.PropsWithChildren<
        | { loading: true }
        | {
          loading?: false;
          currency: CurrencyData;
          selected?: boolean;
          onClick?: () => void;
        }
      >
    >;
  };
  groupHeight?: number;
  itemHeight?: number;
  maxHeight?: number;
  embed?: boolean;
};
