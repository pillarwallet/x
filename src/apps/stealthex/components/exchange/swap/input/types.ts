
import { CurrencyData } from '../../../../lib/backend/api';
import type { AvailableCurrency } from '../dropdown/normal/types';

export type SavedAvailableCurrency =
  | {
      floatingDropdown: true;
      availableCurrency: CurrencyData[];
      type: 'get' | 'send';
    }
  | ({
      floatingDropdown?: false;
    } & AvailableCurrency);

export type InputProps = {
  currency: CurrencyData;
  value?: string;
  loading?: boolean;
  loadingCurrency?: boolean;
  error?: string | JSX.Element;
  fiat?: boolean;
  disabled?: boolean;
  withLock?: boolean | null;
  widget?: boolean;
  fixed?: boolean;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  onFixedChange?: (fixed: boolean) => void;
  onCurrencySelect?: (currency: CurrencyData) => void;
  pressable?: boolean;
} & SavedAvailableCurrency;
