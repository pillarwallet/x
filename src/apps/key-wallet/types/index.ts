export interface Asset {
  id: number;
  name: string;
  symbol: string;
  logo: string;
  balance: number;
  decimals: number;
  price: number;
  price_change_24h: number;
  contract: string;
  chainId: number;
  chainName: string;
  usdBalance: number;
}

export interface SendFormData {
  amount: string;
  recipient: string;
}

export interface TransactionStatus {
  hash: string;
  chainId: number;
  status: 'pending' | 'success' | 'failed';
  timestamp: number;
}

