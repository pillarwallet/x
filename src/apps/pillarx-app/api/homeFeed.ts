/* eslint-disable no-console */
import { createApi, fetchBaseQuery, retry } from '@reduxjs/toolkit/query/react';

// store
import { addMiddleware } from '../../../store';

// types
import { ApiResponse, WalletData } from '../../../types/api';

// utils
import { CompatibleChains } from '../../../utils/blockchain';

const chainIds =
  process.env.REACT_APP_USE_TESTNETS === 'true'
    ? [11155111]
    : CompatibleChains.map((chain) => chain.chainId);

const chainIdsQuery = chainIds.map((id) => `chainIds=${id}`).join('&');

export const homeFeedApi = createApi({
  reducerPath: 'homeFeedApi',
  baseQuery: fetchBaseQuery({
    baseUrl:
      process.env.REACT_APP_USE_TESTNETS === 'true'
        ? 'https://feed-nubpgwxpiq-uc.a.run.app'
        : 'https://feed-7eu4izffpa-uc.a.run.app',
  }),
  endpoints: (builder) => ({
    getTilesInfo: builder.query<ApiResponse, { page: number; address: string }>(
      {
        query: ({ page, address }) =>
          `?page=${page}&address=${address}&${chainIdsQuery}&testnets=${process.env.REACT_APP_USE_TESTNETS || 'true'}`,
      }
    ),
  }),
});

// maxRetries: 5 is the default, and can be omitted. Shown for documentation purposes.
const staggeredBaseQuery = retry(
  fetchBaseQuery({
    baseUrl:
      process.env.REACT_APP_USE_TESTNETS === 'true'
        ? 'https://walletportfolio-nubpgwxpiq-uc.a.run.app'
        : 'https://walletportfolio-7eu4izffpa-uc.a.run.app',
  }),
  {
    maxRetries: 5,
  }
);

export const walletPortfolioTileApi = createApi({
  reducerPath: 'walletPortfolioTileApi',
  baseQuery: staggeredBaseQuery,
  endpoints: (builder) => ({
    getWalletInfo: builder.query<WalletData, { address: string }>({
      query: ({ address }) =>
        `?address=${address}&${chainIdsQuery}&testnets=${process.env.REACT_APP_USE_TESTNETS || 'true'}`,
    }),
  }),
});

/**
 * Add this to the store
 */
addMiddleware(homeFeedApi);
addMiddleware(walletPortfolioTileApi);

export const { useGetTilesInfoQuery } = homeFeedApi;
export const { useGetWalletInfoQuery } = walletPortfolioTileApi;
