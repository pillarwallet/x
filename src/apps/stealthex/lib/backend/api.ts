import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
import { RequestResponse } from './type';
import { ExchangeInfo } from '../../type';
import { ProviderName } from '../providers';

export const baseUrl = 'https://stealthex.io';

export type Abortable = { signal?: AbortSignal }

export type IsFixed = {
    isFixed: boolean;
};

export type CreateExchangeParams = {
    currencyFrom: string;
    currencyTo: string;
    address: string;
    extraId?: string;
    amountTo?: string;
    amountFrom?: string;
    referral?: string;
    fixed?: boolean;
    timezoneOffset: number;
    isWidget?: boolean;
    partnerId?: string;
    rate_id?: string;
    refundAddress?: string;
    provider?: ProviderName;
};
export type Estimate = {
    estimate: string | null;
    rate_id?: string;
};

export type Range = {
    min_amount?: string;
    max_amount?: string;
    message?: string;
};

export type EstimateProps = {
    from: string | null;
    to: string | null;
    amount?: number;
    amount_to?: number;
    fixed: boolean;
    provider?: 'simplex' | 'mercuryo';
    signal?: AbortSignal;
};

export type RangeProps = {
    to: string | null;
    from: string | null;
    fixed?: boolean;
    fiat?: boolean;
    reverse?: boolean;
};

export type CoinImage = {
    id?: string;
    mimetype?: string;
    size?: number;
    filename: string;
    originalFilename?: string;
    encoding?: string;
    _meta?: boolean;
};

export type Post = {
    id: number;
    date: string;
    link: string;
    title: {
        rendered: string;
    };
    excerpt: {
        rendered: string;
    };
    featured_media: number;
    categories: string[];
    tags: number[];
    modified?: string;
};

export type FormattedPost = Post & {
    image: string;
};

export type CoinData = {
    change24h?: string;
    changepct24h?: string;
    high24h?: string;
    low24h?: string;
    mktcap?: string;
    price: string;
    network: string | null;
    supply?: string;
    volume24h?: string;
    _id: string;
    image: string;
    symbol: string;
    slug: string;
    name: string;
    about?: string;
    order?: number;
    colorIcon?: CoinImage;
    seoImage?: CoinImage;
    disableTradingView: false;
    priceChart: {
        type: 'third_party';
        symbol: string;
    };
    relatedArticles: FormattedPost[];
};

export type CurrencyData = {
    symbol: string;
    slug?: string;
    has_extra_id: boolean;
    extra_id: string | null;
    is_fiat?: boolean;
    name: string;
    network: string | null;
    warnings_from: string[];
    warnings_to: string[];
    validation_address: string | null;
    validation_extra: string | null;
    address_explorer: string | null;
    tx_explorer: string | null;
    image: string;
};

export type PairsTo = {
    popular: CurrencyData[];
    other: CurrencyData[];
};

export type PairsFrom = {
    popular: CurrencyData[];
    other: CurrencyData[];
    fiat: CurrencyData[];
};

export const exchange = createApi({
    reducerPath: 'exchangeInfo',
    baseQuery: fetchBaseQuery({ baseUrl }),
    endpoints: (builder) => ({
        exchangeInfo: builder.query<{ data: ExchangeInfo, signature: string }, string>({
            query: (exchangeId) => ({ url: `/api/exchange/v2/${exchangeId}` })
        }),
        createExchange: builder.mutation<{ data: ExchangeInfo, signature: string }, CreateExchangeParams>({
            query: ({ ...params }) => {
                const { currencyFrom, currencyTo, address, extraId, amountFrom, amountTo, isWidget, partnerId, refundAddress, ...otherParams } = params;
                return ({
                    url: '/api/exchange/v2/create', method: 'POST', body: {
                        currency_from: currencyFrom,
                        currency_to: currencyTo,
                        address_to: address,
                        extra_id_to: extraId,
                        amount_from: amountFrom,
                        amount_to: amountTo,
                        partner_id: partnerId,
                        is_widget: isWidget,
                        refund_address: refundAddress,
                        refund_extra_id: '',
                        ...otherParams,
                    }
                })
            },
        }),
        estimate: builder.query<Estimate, EstimateProps>({
            query: ({ amount, amount_to, from, to, fixed, signal }) => ({ url: `/api/exchange/estimate/v3/${from}/${to}`, params: { amount, fixed, amount_to }, signal })
        }),
        fiatEstimate: builder.query<Estimate, EstimateProps>({
            query: ({ amount, from, to, signal, provider }) => ({ url: `/api/exchange/estimate/v3/${from}/${to}`, params: { amount, fixed: false, provider }, signal })
        }),
        fiatCoins: builder.query<RequestResponse<false, CoinData[]>, Abortable>({
            query: ({ signal }) => ({ url: '/api/coins/fiat', signal, params: { timeout: false } })
        }),
        searchCurrency: builder.query<RequestResponse<false, CurrencyData>, { query: string, signal?: AbortSignal, onlySymbolSearch: boolean, }>({
            query: ({ signal, query, onlySymbolSearch }) => ({ url: '/api/exchange/searchCurrency', signal, params: { query, only_symbol_search: onlySymbolSearch } })
        }),
        range: builder.query<Range, RangeProps & { signal?: AbortSignal, enabled?: boolean }>({
            query: ({ to, from, signal, ...searchParams }) => ({ url: `/api/exchange/range/${from}/${to}`, params: searchParams, signal })
        }),
        pairsFrom: builder.query<PairsFrom, { coin: string | null } & Abortable>({
            query: ({ coin, signal }) => ({ url: `/api/exchange/pairsfrom/${coin}`, signal })
        }),
        pairsTo: builder.query<PairsTo, { coin: string | null } & Abortable>({
            query: ({ coin, signal }) => ({ url: `/api/exchange/pairs/${coin}`, signal })
        }),
        isFixed: builder.query<IsFixed, { from: string, to: string } & Abortable>({
            query: ({ from, to, signal }) => ({ url: `/api/exchange/is-fixed/${from}/${to}`, signal })
        })
    })
})

// export const estimate = createApi({
//     reducerPath: 'estimate',
//     baseQuery: fetchBaseQuery({baseUrl}),
//     endpoints: (builder) => ({
//         estimateInfo: builder.mutation<RequestResponse<true,Estimate>,CreateExchangeParams & {signal: AbortSignal}>({
//             query: ({currencyTo, currencyFrom,signal,  ...searchParams }) => ({url: `/api/exchange/estimate/v3/${currencyFrom}/${currencyTo}/?amount=${searchParams.}`,signal, method: 'POST'  })
//         })
//     })
// })


export const { useExchangeInfoQuery, useCreateExchangeMutation, useFiatEstimateQuery, useFiatCoinsQuery, useSearchCurrencyQuery, useRangeQuery, useEstimateQuery, usePairsFromQuery, usePairsToQuery, useIsFixedQuery } = exchange;