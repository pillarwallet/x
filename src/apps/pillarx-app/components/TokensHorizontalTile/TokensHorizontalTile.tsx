import { createRef, useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';

// types
import { Projection, TokenData } from '../../../../types/api';

// hooks
import useRefDimensions from '../../hooks/useRefDimensions';

// components
import TileContainer from '../TileContainer/TileContainer';
import TokenInfoHorizontal from '../TokenInfoHorizontal/TokenInfoHorizontal';
import Body from '../Typography/Body';

type TokensHorizontalTileProps = {
  data: Projection | undefined;
  isDataLoading: boolean;
};

const TokensHorizontalTile = ({
  data,
  isDataLoading,
}: TokensHorizontalTileProps) => {
  const navigate = useNavigate();
  const [tokenHorizontalWidth, setTokenHorizontalWidth] = useState<number>(0);
  const { data: dataTokens, meta } = data || {};
  const dataTokensHorizontal = dataTokens as TokenData[] | undefined;

  const windowWidth = window.innerWidth;

  useEffect(() => {
    const handleTokenHorizontalWidthResize = () => {
      if (windowWidth >= 1024) {
        setTokenHorizontalWidth(159);
      } else if (windowWidth >= 800) {
        setTokenHorizontalWidth(152);
      } else {
        // 102px for width of the TokenInfoHorizontal component 100px + 2px of space
        setTokenHorizontalWidth(102);
      }
    };

    handleTokenHorizontalWidthResize();
    window.addEventListener('resize', handleTokenHorizontalWidthResize);

    return () => {
      window.removeEventListener('resize', handleTokenHorizontalWidthResize);
    };
  }, [windowWidth]);

  const divRef = createRef<HTMLDivElement>();
  const dimensions = useRefDimensions(divRef);

  const numberTokensHorizontal =
    Math.floor(dimensions.width / tokenHorizontalWidth) ?? 0;

  if (!data || isDataLoading) {
    return null;
  }

  return (
    <div ref={divRef}>
      <TileContainer
        id="tokens-horizontal-tile"
        className="flex-col px-10 pt-[30px] pb-5 tablet:p-5 mobile:p-0 mobile:bg-[#1F1D23]"
      >
        <Body className="text-purple_light mb-2.5">{meta?.display?.title}</Body>
        <div className="flex justify-between">
          {dataTokensHorizontal
            ?.slice(0, numberTokensHorizontal)
            .map((token, index) => (
              <TokenInfoHorizontal
                key={index}
                logo={token.logo}
                tokenName={token.name}
                tokenValue={undefined}
                percentage={undefined}
                onClick={() =>
                  navigate(
                    `/token-atlas?asset=${token.name}&symbol=${token.symbol}`
                  )
                }
              />
            ))}
        </div>
      </TileContainer>
    </div>
  );
};

export default TokensHorizontalTile;
