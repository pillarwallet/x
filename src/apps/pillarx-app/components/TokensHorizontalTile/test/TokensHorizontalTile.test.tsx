import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

// components
import { ApiLayout, Projection } from '../../../../../types/api';
import TokensHorizontalTile from '../TokensHorizontalTile';

describe('<TokensHorizontalTile />', () => {
  const mockData: Projection = {
    meta: {
      display: {
        title: 'horizontal title',
      },
    },
    data: [
      {
        id: 1,
        name: 'The best token',
        symbol: 'TBT',
        contracts: [
          {
            address: '0xD6DF932A45C0f255f85145f286eA0b292B21C90B',
            blockchain: '137',
            decimals: 18,
          },
        ],
        logo: 'https://www.logo.com/example/',
        trending_score: 20,
        platforms: [
          {
            name: 'platform one',
            rank: 3,
            weight: 1,
          },
        ],
        price_change_24h: 12.56,
        pair: 'pair example',
      },
    ],
    layout: ApiLayout.TOKENS_HORIZONTAL,
    id: 'horizontal',
  };

  const mockLoading = false;

  it('renders correctly and matches snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <TokensHorizontalTile data={mockData} isDataLoading={mockLoading} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('displays loading skeleton when data is loading', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <TokensHorizontalTile data={undefined} isDataLoading />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('returns null when dataTokens is empty', () => {
    const emptyMockData: Projection = {
      meta: {
        display: {
          title: 'horizontal title',
        },
      },
      data: [],
      layout: ApiLayout.TOKENS_HORIZONTAL,
      id: 'horizontal',
    };

    const tree = renderer
      .create(
        <MemoryRouter>
          <TokensHorizontalTile
            data={emptyMockData}
            isDataLoading={mockLoading}
          />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toBeNull();
  });
});
