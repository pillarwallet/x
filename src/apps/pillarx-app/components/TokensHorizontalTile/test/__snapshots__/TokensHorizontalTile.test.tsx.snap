// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokensHorizontalTile /> displays loading skeleton when data is loading 1`] = `null`;

exports[`<TokensHorizontalTile /> renders correctly and matches snapshot 1`] = `
<div>
  <div
    className="flex bg-container_grey rounded-2xl flex-col px-10 pt-[30px] pb-5 tablet:p-5 mobile:p-0 mobile:bg-[#1F1D23]"
    id="tokens-horizontal-tile"
  >
    <p
      className="text-base font-medium text-purple_light mb-2.5"
    >
      horizontal title
    </p>
    <div
      className="flex justify-between"
    >
      <div
        className="flex flex-col py-5 px-[22px] gap-1 w-[122px] h-auto items-center cursor-pointer tablet:w-[120px] mobile:w-[100px] mobile:px-3.5"
        onClick={[Function]}
      >
        <img
          alt="token-logo"
          className="w-[70px] h-[70px] object-fill rounded-full mb-3.5"
          src="https://www.logo.com/example/"
        />
        <p
          className="text-base font-medium text-center"
        >
          The best token
        </p>
      </div>
    </div>
  </div>
</div>
`;
